File: split/sprintcs.pas
procedure sprintcs(p: halfword);
begin
  if p < 514 then
    if p < 257 then
      print(p - 1)
    else if p < 513 then
      printesc(
        p - 257)
    else
    begin
      printesc(504);
      printesc(505);
    end
  else
    printesc(hash[p].rh);
end;

File: cpp/sprintcs.cpp
#include "sprintcs.h"
#include "printesc.h"
#include "print.h"

void sprintcs(halfword p)
{
	if (p < 514)
		if (p < 257)
			print(p-1);
		else 
			if (p < 513)
				printesc(p-257);
			else
			{
				printesc(504); //csname
				printesc(505); //endcsname
			}
	else
		printesc(hash[p].rh);
}
